orm/orm.go

```diff
@@ -209,6 +209,19 @@ func (o *orm) InsertMulti(bulk int, mds interface{}) (int64, error) {
 	return cnt, nil
 }
 
+// InsertOrUpdate data to database
+func (o *orm) InsertOrUpdate(md interface{}, colConflitAndArgs ...string) (int64, error) {
+	mi, ind := o.getMiInd(md, true)
+	id, err := o.alias.DbBaser.InsertOrUpdate(o.db, mi, ind, o.alias.TZ, o.alias.DriverName, colConflitAndArgs...)
```

Comment:
o.alias.DriverName change to o.alias.DriverType. As we already has define https://github.com/astaxie/beego/blob/master/orm/db_alias.go#L29-L36. So you can remove the hardcode mysql and postgres. As sometime the DriverName is not equal to mysql and postgres.